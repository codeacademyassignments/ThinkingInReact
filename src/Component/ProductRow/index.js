import React from 'react';

const ProductRow = (props)=>{
const name = props.inStock ? props.name : <span id='name' style={{color:'red'}}>{props.name}</span>;
return (<tr>
  <td >{name}</td>
  <td> {props.price}</td>
  </tr>);
}

export default ProductRow;