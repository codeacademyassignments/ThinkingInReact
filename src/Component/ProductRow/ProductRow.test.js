import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import React from 'react';
//import { isMainThread } from 'worker_threads';
import ProductRow from './index';

describe('ProductRow',()=>{
  
  it('snapshot testing',()=>{ 
    const tree = renderer.create(<ProductRow name='football' price='$45' inStock={true}/>);
    expect(tree).toMatchSnapshot();
  })
  it('should red the text passed',()=>{
    const wrapper = shallow(<ProductRow name='football' price = '$45' inStock = {false}/>);
    expect(wrapper.find('#name').prop('style')).toHaveProperty('color','red');
  })
  it('should red the text passed',()=>{
    const wrapper = shallow(<ProductRow name='football' price = '$45' inStock = {true}/>);
    expect(wrapper.find('#name').length).toEqual(0);
  })
  // it('should set its props to name and price',()=>{
  //   const wrapper = shallow(<ProductRow name='football' price='$45'/>);
  //   console.log('herer');
  //   console.log(wrapper.instance.props);
  //   expect(wrapper.find('.Statue').props().name).toEqual('football');
  //   expect(wrapper.props().price).toEqual('$45');
  // })
})