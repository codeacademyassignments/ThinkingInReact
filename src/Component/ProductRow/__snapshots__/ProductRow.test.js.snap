// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRow snapshot testing 1`] = `
<tr>
  <td>
    football
  </td>
  <td>
     
    $45
  </td>
</tr>
`;
